src/main.ts

```diff
@@ -51,16 +52,20 @@ export function deactivate() {
 //   subscriptions.add(busySignalProvider)
 // }
 
-export async function consumeOutlineProvider(provider: OutlineProvider) {
-  subscriptions.add(/*  providerRegistryEntry */ outlineProviderRegistry.addProvider(provider))
+export function consumeOutlineProvider(provider: OutlineProvider): Disposable {
+  const prividerDisposable = outlineProviderRegistry.addProvider(provider)
+  subscriptions.add(/*  providerRegistryEntry */ prividerDisposable)
 
   // NOTE Generate (try) an outline after obtaining a provider for the current active editor
   // this initial outline is always rendered no matter if it is visible or not,
   // this is because we can't track if the outline tab becomes visible suddenly later,
   // or if the editor changes later once outline is visible
   // so we need to have an outline for the current editor
   // the following updates rely on the visibility
-  await getOutline()
+  getOutline().catch((err) => {
+    throw err 
```

Comment:
What are you trying to do here? The function is still executed in promise context, rethrowing the error is basically a no-op. I think you meant to call `atom.notifications.addFatalError` or something similar.